packages/core-database/lib/wallet-manager.js

```diff
@@ -376,6 +376,41 @@ module.exports = class WalletManager {
     return data
   }
 
+  /**
+   * Determine the list of active delegates for the given round.
+   * @param  {Number} maxDelegates
+   * @return {Array}
+   */
+  determineActiveDelegates (maxDelegates) {
+    const mapWallets = wallets => wallets.map(wallet => ({
+      publicKey: wallet.vote || wallet.publicKey,
+      balance: wallet.balance ? wallet.balance.toFixed() : 0
+    }))
+
+    let delegates = mapWallets(this.allByPublicKey().filter(wallet => !!wallet.vote))
```

Comment:
Maybe I miss some edge case, but I think the use of `!!` here is just an obfuscation. I think the following two are identical:

```
.filter(wallet => !!wallet.vote))
```
and
```
.filter(wallet => wallet.vote))
```